Source/DafnyLanguageServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -879,12 +859,12 @@ public class DafnyModel {
     }
 
     /// <summary>
-    /// Return all functions that map the datatype object to a particular
-    /// destructor value.
+    /// Return all functions that map an object to a destructor value.
     /// </summary>
     private List<Model.Func> GetDestructorFunctions(Model.Element datatypeElement) {
       var types = GetIsResults(datatypeElement).Select(isResult =>
         new DafnyModelTypeUtils.DatatypeType(ReconstructType(isResult) as UserDefinedType ?? UnknownType).Name);
+      types = types.Select(type => type.Length > 0 && type.Last() == '?' ? type[..^1] : type);
```

Comment:
Is it possible to chop off the `?` at the type level instead of at the string level? Something like `Type.NonNullable`

Could you put the "?" into a const variable?

Could you replace the reassignment with a fresh assignment?

For example using,

```
var possiblyNullableTypes = ..; 
var types = possiblyNullableTypes.Select(type => type.Length > 0 && type.Last() == '?' ? type[..^1] : type)
```

Reply:
Done - changed the code to use `DafnyModelTypeUtils.GetNonNullable` (I don't think there is API to perform this operation in the Type class itself, at least not on the unresolved level)